packages/lib/getCalendarsEvents.ts

```diff
@@ -71,8 +109,22 @@ const getCalendarsEvents = async (
       .filter((sc) => sc.integration === type)
       // Needed to ensure cache keys are consistent
       .sort((a, b) => (a.externalId < b.externalId ? -1 : a.externalId > b.externalId ? 1 : 0));
-
-    if (!passedSelectedCalendars.length) return [];
+    const credential = calendarToCredentialMap.get(calendarService);
+    const isADelegationCredential = credential && isDelegationCredential({ credentialId: credential.id });
+    // We want to fallback to primary calendar when no selectedCalendars are passed
+    // Default behaviour for Google Calendar is to use all available calendars, which isn't good default.
+    const allowFallbackToPrimary = isADelegationCredential;
+    if (!passedSelectedCalendars.length) {
+      if (!isADelegationCredential) {
+        // It was done to fix the secondary calendar connections from always checking the conflicts even if intentional no calendars are selected.
+        // https://github.com/calcom/cal.com/issues/8929
+        log.error("No selected calendars for non DWD credential: Skipping getAvailability call");
+        return [];
+      }
+      // For delegation credential, we should allow getAvailability even without any selected calendars. It ensures that enabling Delegation Credential at Organization level always ensure one selected calendar for conflicts checking, without requiring any manual action from organization members
+      // This is also, similar to how Google Calendar connect flow(through /googlecalendar/api/callback) sets the primary calendar as the selected calendar automatically.
```

Comment:
Had to duplicate it because it is in a duplicate fn(`getCalendarsEventsWithTimezone`) that duplicates a lot of `getCalendarsEvents` fn code